RxCocoa/Foundation/NSObject+Rx+RawRepresentable.swift

```diff
@@ -47,6 +81,40 @@ extension Reactive where Base: NSObject {
             return observeWeakly(E.RawValue.KVOType.self, keyPath, options: options)
                 .map(E.init)
         }
+
+        /**
+         Specialization of generic `observeWeakly` method.
+
+         This specialization first observes `KVORepresentable` value and then converts it to `RawRepresentable` value.
+
+         It is useful for observing bridged ObjC enum values.
+
+         For more information take a look at `observeWeakly` method.
+         */
+        @available(swift 4.0)
+        public func observeWeakly<E: RawRepresentable>(_ keyPath: KeyPath<Base, E>, options: KeyValueObservingOptions = [.new, .initial]) -> Observable<E?> where E.RawValue: KVORepresentable {
```

Comment:
I think this signature and other `observeWeakly` are correct because `nil` will be emitted if the object is `nil`.

I guess if somebody has `KeyPath<Base, E?>` they could easily do `observeWeakly.map { $0.flatMap { $0 }}` :)